frontends/election-manager/src/app_root.tsx

```diff
@@ -254,20 +253,26 @@ export function AppRoot({
     );
   }, [storage]);
 
-  async function savePrintedBallots(printedBallotsToStore: PrintedBallot[]) {
-    await setStorageKeyAndLog(
-      printedBallotsStorageKey,
-      printedBallotsToStore,
-      'Printed ballot information'
-    );
-  }
+  const savePrintedBallots = useCallback(
+    async (printedBallotsToStore: PrintedBallot[]) => {
+      await setStorageKeyAndLog(
+        printedBallotsStorageKey,
+        printedBallotsToStore,
+        'Printed ballot information'
+      );
+    },
+    [setStorageKeyAndLog]
+  );
 
-  async function addPrintedBallot(printedBallot: PrintedBallot) {
-    const ballots = await getPrintedBallots();
-    ballots.push(printedBallot);
-    await savePrintedBallots(ballots);
-    setPrintedBallots(ballots);
-  }
+  const addPrintedBallot = useCallback(
```

Comment:
> In in any case, I traced a lot of spurious re-renders to a callback function addPrintedBallots that this component gets from AppContext and uses to log official ballots were printed. That function was being created on every render, so I replaced it with a useCallback. Dependencies forced me to make a number of other functions useCallbacks, so I just made all the callbacks I found in app_root.tsx into useCallbacks. This reduced the re-renders I was seeing and I have not been able to reproduce the issue since.

Glad that this fixed our downstream issues for now! And this seems totally reasonable as a temporary fix for m14. That said, I could definitely see someone down the road removing the `useCallback` wrappings in this file, not realizing that they're relied on downstream.

Could we apply a temporary fix further downstream, like we do for test deck printing?

https://github.com/votingworks/vxsuite/blob/80984f19c14e39e8303fee2972ee33f420a01ecb/frontends/election-manager/src/screens/print_test_deck_screen.tsx#L246-L258

Reply:
I tried memoizing the component and it didn't do anything. I didn't devote the time into digging into why! I think it's because in this case, the prop value passed to `React.memo` does in fact change (the callback from `AppContext`) so it is re-rendered anyway. I think we have to memoize further upstream.

That being said, I understand the concern. Upstream memoization is a recipe for someone hitting this same issue later.

Reply:
Got it! In that case, let's just open a ticket to document that we want to find a more robust solution after m14